content/testable-architecture.md

```diff
@@ -3,4 +3,62 @@ title: Testable Architecture
 description: Designing for testability
 ---
 
-Testable Architecture
\ No newline at end of file
+Automated tests and executable specifications can only work with software that has a testable architecture.
+A testable architecture enables fast and reliable tests that are easy to write, execute and maintain.
+Fast tests enable the developers to run them and obtain fast feedback on what they are building, without losing focus or flow.
+
+When designing for testability, make sure that your products and services are composed of loosely-coupled,
+well-encapsulated components or modules; following the principles of
+[SOLID design](https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)).
+
+Decoupling your business logic from your infrastructure (the often slow and brittle components),
+makes it possible to test at different levels, with maximum confidence and minimum cost.
+
+* Ensure that you can run your tests underneath your UI. Do not depend solely on UI tests. They are slow, expensive, and hard to fix.
+* Running tests against a database makes tests slow. And to make the tests behave consistently each time we run them,
+we need to make sure the database is in the expected state for each test, so the tests don't interfere with one another.
+
```

Comment:
I would turn around this sentence: 

"And to make the tests behave consistently each time we run them, we need to make sure the database is in the expected state for each test, so the tests don't interfere with one another."

Something like this:

We need to make sure the database is in the expected state for each test in order to make the tests behave consistently. This will ensure that the tests don't interfere with each other and can be executed in any order.